src/pages/workspaces/workspace/Data.js

```diff
@@ -59,29 +93,87 @@ class WorkspaceData extends Component {
     this.refresh()
   }
 
+  render() {
+    const { selectedDataType, entityMetadata, loading } = this.state
+    const { namespace, name } = this.props
+
+    return h(WorkspaceContainer, {
+      namespace, name,
+      refresh: () => this.refresh(),
+      breadcrumbs: breadcrumbs.commonPaths.workspaceDashboard({ namespace, name }),
+      title: 'Data', activeTab: 'data'
+    }, [
+      div({ style: styles.tableContainer }, [
+        !entityMetadata ? spinnerOverlay : h(Fragment, [
+          div({ style: styles.dataTypeSelectionPanel }, [
+            div({ style: styles.dataModelHeading }, 'Data Model'),
+            !_.isEmpty(entityMetadata) && div({ style: { borderBottom: `1px solid ${Style.colors.background}` } },
+              _.map(([type, typeDetails]) =>
+                div({
+                  style: styles.dataTypeOption(selectedDataType === type),
+                  onClick: () => {
+                    this.setState({ selectedDataType: type, pageNumber: 1, refreshRequested: true })
```

Comment:
This needs to set `entities` to undefined, or else it shows the old data under the new column headings. This will also make `refreshRequested` unnecessary.

Reply:
Still need the `refreshRequested`. `filterState` doesn't look at data any more, that's what was causing the double load.